### PR TITLE
fix: #768 - fix smb app - page: office, negotiator

### LINKED ISSUE #768: Amendments to Offices Page of SMB App
**Summary:** Enhancements and bug fixes required to Offices Page of SMB App to improve UX & some UI

**Location:** https://dev.smb-app.reapit.cloud/Offices

**Task:**

Amendment to the 'Download File' feature – 
- When I select 'download file', it only downloads the first page. (12 records). This should show all Office data.
- 'Office reference' column is not being downloaded
- 'Building name' currently pulling through as code – 'Building name' to be pulled through
 
image

- Postcode, Telephone & Email in incorrect column format/alignment; Column alignment to be corrected so correct information showing in relevant column.
 
image

- After I clicked 'upload file' & clicked Next Page - The below error was shown:
image

 
- On the Office table, if there is an error on any single row, any additional row(s) should not be editable, allowing error(s) only on one line at a time.
 
image

- Email column to fit wording within field – Wrap
image

- Email column – Rename 'Email' to ‘Office Email’

### PR DESCRIPTION
**Pull request by phmngocnghia**
Fixes #768

### CODE REVIEW

File: packages/elements/src/components/Spreadsheet/__tests__/generateDataWithReadOnlyAndIsValidated.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import * as utils from '../utils'
```
Comment (willmcvay) [inline packages/elements/src/components/Spreadsheet/__tests__/generateDataWithReadOnlyAndIsValidated.ts:1]: Re-adding my comment about kebab-case filenames

File: packages/elements/src/components/Spreadsheet/utils.tsx [hunk 2]
```diff
@@ -91,10 +91,96 @@ export const changedCellsGenerate = (newData?: Cell[][], oldData?: Cell[][]): Ch
   return changedCells
 }
 
+/**
```
Comment (willmcvay) [inline packages/elements/src/components/Spreadsheet/utils.tsx:94]: Thankyou, this is much better

File: packages/smb-onboarder/src/components/ui/negotiators-list/__mocks__/negotiators.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { NegotiatorsQueryResponse, NegotiatorUpdateParams, NegotiatorCreateParams } from '../negotiators-list'
-import { NegotiatorModel } from '@reapit/foundations-ts-definitions/types'
+import { NegotiatorModel } from '@reapit/elements/node_modules/@reapit/foundations-ts-definitions/types'
```
Comment (vuhuucuong) [inline packages/smb-onboarder/src/components/ui/negotiators-list/__mocks__/negotiators.ts:2]: Why do we need to import like this?

Comment (phmngocnghia) [reply packages/smb-onboarder/src/components/ui/negotiators-list/__mocks__/negotiators.ts:2]: In the original commit (`08884de6a8d70bbd9ccfed34b36ce026121fc881`). I saw it was written like this... Seems like other commit has addressed this issue

File: packages/elements/src/components/Spreadsheet/handlers.tsx [hunk 4]
```diff
@@ -98,7 +109,7 @@ export const customCellRenderer = (
     'updated',
     'style',
   ]
-  // filter out all ignore props
+  // filter ouf all ignore props
```
[the remark this answers is not in the document]
Comment (phmngocnghia) [reply packages/elements/src/components/Spreadsheet/handlers.tsx:112]: Thanks man. Nice catch!

**Pull request merged**